brokerapi/brokers/account_managers/service_account_manager.go

```diff
@@ -184,6 +161,82 @@ func ServiceAccountName(bindingId string) string {
 	}
 }
 
+func (sam *ServiceAccountManager) createServiceAccount(bindingID string) (*iam.ServiceAccount, error) {
+	client := sam.HttpConfig.Client(context.Background())
+	iamService, err := iam.New(client)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating new IAM service: %s", err)
+	}
+
+	someName := ServiceAccountName(bindingID)
+	resourceName := projectResourcePrefix + sam.ProjectId
+
+	// create and save account
+	newSARequest := iam.CreateServiceAccountRequest{
+		AccountId: someName,
+		ServiceAccount: &iam.ServiceAccount{
+			DisplayName: someName,
+		},
+	}
+
+	return iam.NewProjectsServiceAccountsService(iamService).Create(resourceName, &newSARequest).Do()
+}
+
+func (sam *ServiceAccountManager) createServiceAccountKey(account *iam.ServiceAccount) (*iam.ServiceAccountKey, error) {
+	client := sam.HttpConfig.Client(context.Background())
+	iamService, err := iam.New(client)
+	if err != nil {
+		return nil, fmt.Errorf("Error creating new IAM service: %s", err)
+	}
+
+	saKeyService := iam.NewProjectsServiceAccountsKeysService(iamService)
+	return saKeyService.Create(account.Name, &iam.CreateServiceAccountKeyRequest{}).Do()
+}
+
+func (sam *ServiceAccountManager) grantRoleToAccount(role string, account *iam.ServiceAccount) error {
+	client := sam.HttpConfig.Client(context.Background())
+
+	cloudresService, err := cloudres.New(client)
+	if err != nil {
+		return fmt.Errorf("Error creating new cloud resource management service: %s", err)
+	}
+
+	currPolicy, err := cloudresService.Projects.GetIamPolicy(sam.ProjectId, &cloudres.GetIamPolicyRequest{}).Do()
+	if err != nil {
+		return fmt.Errorf("Error getting current project iam policy: %s", err)
+	}
+
+	// seems not really necessary, but collapse the bindings into single role entries just in case.
+	var existingBinding *cloudres.Binding
+
+	for _, binding := range currPolicy.Bindings {
+		if binding.Role == roleResourcePrefix+role {
+			existingBinding = binding
+		}
+	}
+
+	if existingBinding != nil {
+		existingBinding.Members = append(existingBinding.Members, saResourcePrefix+account.Email)
```

Comment:
This might be one where we should pull out the code from the Terraform Google Provider and inject it here. Someone over there has written code to handle this odd case. I'll check it out tomorrow.